pkg/api/validate/dynamic/dynamic_test.go

```diff
@@ -450,23 +478,25 @@ func TestValidateCIDRRanges(t *testing.T) {
 				tt.modifyOC(oc)
 			}
 
-			vnetClient := mock_network.NewMockVirtualNetworksClient(controller)
-			if tt.vnetMocks != nil {
-				tt.vnetMocks(vnetClient, vnet)
-			}
+			for _, vnet := range vnets {
```

Comment:
I see that we loop over few flavours of vnet definition, but I think failure in this loop will be hard to debug.

I would add another another `t.Run` somewhere here and make sure to give a good name to it.

Something like:

```go
vnet := map[string]mgmtnetwork.VirtualNetwork{
	"vnet with AddressPrefix": {
		ID:       &vnetID,
		// ...
	},
	"vnet with AddressPrefixes": {
		ID:       &vnetID,
		// ...
	},
}

for vnetCase, vnet := range vnets {
	t.Run(vnetCase, func(t *testing.T) {
		// ...
	})
}
```

Reply:
@m1kola Thank you for the input, I added names to the vnets and log those with any error. This should make debugging straight forward + it doesn't require restructuring the whole test.